### DISCUSSION TITLE
Adding and Removing Free Items from Library

### DISCUSSION THREAD

**Discussion by Mbucari** [mkb79/Audible#118] [Show and tell]
~~You may add free items (Audible Plus books, podcast episodes, podcast series, etc) to your library using the following request~~

~~`PUT /1.0/library/item`~~

EDIT* Never mind. I see you already had this in your wiki!

And you may delete free items from your library using the following request

`DELETE /item/[asin]/default_iphone_loan_id`

For both requests, success is indicated with the following response body

```JSON
{
  "error_code": null
}
```

**Comment by mkb79** [comment]
@Mbucari 
Thank you for your posting. I have implement a (private) plugin to add and remove books in `audible-cli` for some month. You have to check for the `is_removable` flag in the library item response. Only if `True` these item can be removed successfully.

**Reply by Mbucari** [reply]
I have not been using `is_removable` to determine if I could remove a book from the library. Instead, I've been using `is_ayce`. So far, it seems to have a 100% success rate. But I will explore `is_removable` more. Thanks!

Another trick I've learned is how to detect if books in your library are unavailable to listen or download. [I explained it here](https://github.com/rmcrackan/Libation/issues/269#issuecomment-1458493288)

**Reply by Mbucari** [reply]
And it's documented here
https://github.com/rmcrackan/AudibleApi/blob/444999e466b980891ca13c993c575f4140933972/AudibleApi.Common/LibraryDtoV10.cs#L85
